src/Figure.tsx

```diff
@@ -3,9 +3,14 @@ import createWithBsPrefix from './createWithBsPrefix';
 import FigureImage from './FigureImage';
 import FigureCaption from './FigureCaption';
 
-const Figure = createWithBsPrefix('figure', {
+type Figure = React.ComponentType & {
+  Image: typeof FigureImage;
+  Caption: typeof FigureCaption;
+};
+
+const Figure: Figure = (createWithBsPrefix('figure', {
   Component: 'figure',
-});
+}) as unknown) as Figure;
```

Comment:
usually we deal with this by using `Object.assign` which handles adjusting the  type